src/lock.ts

```diff
@@ -94,6 +102,10 @@ export class Lock {
     return this.acquire()
       .then(fn)
       .then(value => this.release().then(() => value))
-      .catch(err => this.release().then(() => { throw err; }));
+      .catch(err =>
```

Comment:
I prefer to follow prettier's styling over any of my own personal preferences. Having a single unambiguous style is more important than my collection of individual preferences.